server/entities/game/Game.ts

```diff
@@ -0,0 +1,91 @@
+import GameBuilder from "./game-builder/GameBuilder";
+import { GameState } from "../game-state/GameState";
+
+export default class Game {
+  private state: GameState;
+  private word: string;
+  private matches: string[];
+  private misses: string[];
+
+  constructor(
+    word: string = "",
+    matches: string[] = [],
+    misses: string[] = []
+  ) {
+    this.word = word;
+    this.matches = matches;
+    this.misses = misses;
+    this.state = this.selectState();
+  }
+
+  guess(letter: string): Game {
+    if (this.isLetterAlreadyGuessed(letter)) return this;
+
+    if (this.isGuessCorrect(letter)) return this.addMatch(letter);
+    else return this.addMiss(letter);
```

Comment:
`if (this.isLetterAlreadyGuessed(letter)) return this; return this.isGuessCorrect(letter) ? this.addMatch(letter) : this.addMiss(letter);`